File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/api/MessagesApiImpl.java
package org.openapitools.vertxweb.server.api;

import org.openapitools.vertxweb.server.model.CodedError;
import io.vertx.ext.web.FileUpload;
import org.openapitools.vertxweb.server.model.InvalidMessageError;
import org.openapitools.vertxweb.server.model.JsonSuccess;
import org.openapitools.vertxweb.server.model.JsonSuccessBase;
import org.openapitools.vertxweb.server.model.OneOfobjectobject;
import org.openapitools.vertxweb.server.model.OneOfstringinteger;

import org.openapitools.vertxweb.server.ApiResponse;

import io.vertx.core.Future;
import io.vertx.core.json.JsonObject;
import io.vertx.ext.web.handler.impl.HttpStatusException;

import java.util.List;
import java.util.Map;

// Implement this class

public class MessagesApiImpl implements MessagesApi {
    public Future<ApiResponse<JsonSuccess>> addReaction(Integer messageId, String emojiName, String emojiCode, String reactionType) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<Object>> checkMessagesMatchNarrow(List<Integer> msgIds, List<Object> narrow) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> deleteMessage(Integer messageId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> getFileTemporaryUrl(Integer realmIdStr, String filename) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> getMessageHistory(Integer messageId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> getMessages(Integer numBefore, Integer numAfter, OneOfstringinteger anchor, List<Object> narrow, Boolean clientGravatar, Boolean applyMarkdown, Boolean useFirstUnreadAnchor) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> getRawMessage(Integer messageId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> markAllAsRead() {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> markStreamAsRead(Integer streamId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> markTopicAsRead(Integer streamId, String topicName) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> removeReaction(Integer messageId, String emojiName, String emojiCode, String reactionType) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> renderMessage(String content) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> sendMessage(String type, List<Integer> to, String content, String topic, String queueId, String localId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccess>> updateMessage(Integer messageId, String topic, String propagateMode, Boolean sendNotificationToOldThread, Boolean sendNotificationToNewThread, String content, Integer streamId) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> updateMessageFlags(List<Integer> messages, String op, String flag) {
        return Future.failedFuture(new HttpStatusException(501));
    }

    public Future<ApiResponse<JsonSuccessBase>> uploadFile(FileUpload filename) {
        return Future.failedFuture(new HttpStatusException(501));
    }

}

File: scala-play-server/app/api/MessagesApiImpl.scala
package api

import model.CodedError
import model.InvalidMessageError
import model.JsObject
import model.JsonSuccess
import model.JsonSuccessBase
import model.OasAnyTypeNotMapped
import model.OneOfobjectobject
import model.OneOfstringinteger
import play.api.libs.Files.TemporaryFile

/**
  * Provides a default implementation for [[MessagesApi]].
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2021-08-09T21:37:15.417726Z[Etc/UTC]")
class MessagesApiImpl extends MessagesApi {
  /**
    * @inheritdoc
    */
  override def addReaction(messageId: Int, emojiName: String, emojiCode: Option[String], reactionType: Option[String]): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def checkMessagesMatchNarrow(msgIds: List[Int], narrow: List[JsObject]): OasAnyTypeNotMapped = {
    // TODO: Implement better logic

    null
  }

  /**
    * @inheritdoc
    */
  override def deleteMessage(messageId: Int): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def getFileTemporaryUrl(realmIdStr: Int, filename: String): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def getMessageHistory(messageId: Int): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def getMessages(numBefore: Int, numAfter: Int, anchor: Option[OneOfstringinteger], narrow: Option[List[JsObject]], clientGravatar: Option[Boolean], applyMarkdown: Option[Boolean], useFirstUnreadAnchor: Option[Boolean]): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def getRawMessage(messageId: Int): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def markAllAsRead(): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def markStreamAsRead(streamId: Int): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def markTopicAsRead(streamId: Int, topicName: String): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def removeReaction(messageId: Int, emojiName: Option[String], emojiCode: Option[String], reactionType: Option[String]): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def renderMessage(content: String): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def sendMessage(`type`: String, to: List[Int], content: String, topic: Option[String], queueId: Option[String], localId: Option[String]): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def updateMessage(messageId: Int, topic: Option[String], propagateMode: Option[String], sendNotificationToOldThread: Option[Boolean], sendNotificationToNewThread: Option[Boolean], content: Option[String], streamId: Option[Int]): JsonSuccess = {
    // TODO: Implement better logic

    JsonSuccess(null, null)
  }

  /**
    * @inheritdoc
    */
  override def updateMessageFlags(messages: List[Int], op: String, flag: String): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }

  /**
    * @inheritdoc
    */
  override def uploadFile(filename: Option[TemporaryFile]): JsonSuccessBase = {
    // TODO: Implement better logic

    JsonSuccessBase("", "")
  }
}
